sdcm/nemesis.py

```diff
@@ -1866,6 +1867,34 @@ def modify_table_bloom_filter_fp_chance(self):  # pylint: disable=invalid-name
         """
         self._modify_table_property(name="bloom_filter_fp_chance", val=random.random() / 2)
 
+    def toggle_table_gc_mode(self):
+        """
+            Alters a non-system table tombstone_gc_mode option.
+            Choose the alternate option of 'repair' / 'timeout'
+             (*) The other available values of 'disabled' / 'immediate' are not tested by
+             this nemesis since not applicable to a longevity test.
+        """
+        all_ks_cfs = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node)
+        non_mview_ks_cfs = self.cluster.get_non_system_ks_cf_list(db_node=self.target_node, filter_out_mv=True)
+
+        if not all_ks_cfs:
+            raise UnsupportedNemesis(
+                'Non-system keyspace and table are not found. toggle_table_gc_mode nemesis can\'t run')
+
+        mview_ks_cfs = list(set(all_ks_cfs) - set(non_mview_ks_cfs))
+        keyspace_table = random.choice(all_ks_cfs)
+        keyspace, table = keyspace_table.split('.')
+        if get_gc_mode(node=self.target_node, keyspace=keyspace, table=table) != GcMode.REPAIR:
+            new_gc_mode = GcMode.REPAIR
+        else:
+            new_gc_mode = GcMode.TIMEOUT
+        new_gc_mode_as_dict = {'mode': new_gc_mode.value}
+
+        alter_command_prefix = 'ALTER TABLE ' if keyspace_table not in mview_ks_cfs else 'ALTER MATERIALIZED VIEW '
+        cmd = alter_command_prefix + f" {keyspace_table} WITH tombstone_gc = {new_gc_mode_as_dict};"
+        self.log.debug("Alter GC mode query to execute: %s", cmd)
+        self.target_node.run_cqlsh(cmd)
```

Comment:
Altering gc mode is quite important in terms of a test.
Shouldn't we log it (or even send event) at INFO level?

Reply:
does nemesis event say about new_gc_mode and table that was affected?

Reply:
I think there is no option for such an additional info with current Nemesis implementation. I believe it might be sufficient that only in a case of a nemesis failure and a real scylla issue, more data should be looked at in log file.